analysis/deep2_r23_o32/curvefit_analysis.py

```diff
@@ -8,25 +8,38 @@
 from Zcalbase_gal.analysis.deep2_r23_o32 import bian_coeff, \
     secondorder_polynomial, thirdorder_polynomial, threevariable_fit
 from .plotting import plot_difference_curvefit
+from . import ctype
 from .log_commons import log_stdout, LogClass
 
 
-def fitting_function(lR23_ini, lO32_ini, OH_ini, secondorder=True,
+def fitting_function(lR23_array, lO32_array, OH_array, secondorder=True,
                      threevariable=True):
+    """
+    :param lR23_array: array. an array of arrays of log(R23) values; each array
+                        from a different data set
+    :param lO32_array: array. an array of arrays of log(O32) values; each array
+                        from a different data set
+    :param OH_array: array. an array of arrays of OH values; each array
+                        from a different data set
+    Returns
+    o1, o2 -> contains all parameters from teh curve fitting
+    lR23, lO32, OH -> array of all specific values concatenated and no longer
+                        distinguishable by data sete
+    OH_range -> array of OH values that span the range of OH
+    """
     lR23 = []
     lO32 = []
     OH = []
-    for ii in range(len(lR23_ini)):
-        lR23 = np.concatenate([lR23, lR23_ini[ii]])
-        lO32 = np.concatenate([lO32, lO32_ini[ii]])
-        OH = np.concatenate([OH, OH_ini[ii]])
+    for ii in range(len(lR23_array)):
+        lR23 = np.concatenate([lR23, lR23_array[ii]])
+        lO32 = np.concatenate([lO32, lO32_array[ii]])
+        OH = np.concatenate([OH, OH_array[ii]])
```

Comment:
I think there may be easier ways to collapse a multi-array array into a single array. Better than looping over each element. It's a bit confusing now that I look at it now since lR23,lO32 are defined as lists but they are then soon replaced to be numpy arrays. In fact, I *think* `lR23 = np.concatenate(lR23_array)` should suffice.